### PR TITLE
Improve error handling and error flow

### PR DESCRIPTION
**Pull request by benc-uk**
There were *many* instances of `cobra.CheckErr` in the code, which exits the program by effectively calling `os.Exit(1)`.  
Most of these calls were in lower level utility or helper functions, this makes it hard to understand the flow of errors and also hard to run tests on. 

The main examples being `pkg/azure` and `pkg/terraform` which we should really treat as internal libraries and remove any implementation from them (namely Cobra)

This PR removed most (not all!) of the `cobra.CheckErr` calls from this code level, and modifies the functions to bubble up errors to the Action or CLI level at least, where there can be dealt with

### CODE REVIEW

File: pkg/landingzone/landingzone.go [hunk 1]
```diff
@@ -147,7 +148,9 @@ func (c *TerraformAction) prepareTerraformCAF(o *Options) *tfexec.Terraform {
 func getIndentity(acct azure.Subscription, targetSubID string) azure.Identity {
```
Comment (sebastus) [inline pkg/landingzone/landingzone.go:148]: spelling. getIdentity. Search the repo for 'indent' to get them all. there are 4 or 5.

**Review by sebastus — CHANGES_REQUESTED** [review]
tiny thing.

**Comment by benc-uk** [comment]
> tiny thing.

@sebastus fixed

**Review by sebastus — APPROVED** [review]
lgtm

**Pull request merged**